lib/codegen/CreateTerm.cpp

```diff
@@ -65,8 +66,8 @@ target triple = "x86_64-unknown-linux-gnu"
 %blockheader = type { i64 } 
 %block = type { %blockheader, [0 x i64 *] } ; 16-bit layout, 8-bit length, 32-bit tag, children
 
-%layout = type { i8, %layoutitem* }
-%layoutitem = type { i64, i16 }
+%layout = type { i8, %layoutitem* } ; number of children, array of children
```

Comment:
It looks like these layouts values will be separately named globals, so why does this have a pointer to items rather than directly including the array?

Reply:
This can probably be changed, but I would rather address it separately. Should I create an issue?

Reply:
I asked why you did it this way. If there is a strong reason it would have deserved a comment. If nothing forced this choice then I think a more contiguous layout would probably be more efficient, but I suspect it's not currently a major cost.

I'm definitely not insisting on doing anything without profiling, let alone requesting a specific implementation change - once it matters we should consider the implementation of getLayout more holistically.

Reply:
I think the only reason I did it this way was because it was easier, since it meant that the type returned by getLayoutData and the type of the layout object at allocation could be the same.